hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/validate/HazelcastSqlValidator.java

```diff
@@ -122,6 +124,16 @@ public RelDataType deriveTypeImpl(SqlValidatorScope scope, SqlNode operand) {
                 return literalType;
             }
         }
+        if (operand.getKind() == SqlKind.CASE) {
```

Comment:
Why do we need special handling for the CASE operator? Normally, this should not be the case, unless there is something wrong with Apache Calcite, that forces us to override the behavior.